From 1747a56f89113b8e2900da8e411703025408ea95 Mon Sep 17 00:00:00 2001
From: Kumar-Jy <20044626+Kumar-Jy@users.noreply.github.com>
Date: Sun, 31 Aug 2026 07:13:00 +0530
Subject: [PATCH 52/54] arm64: dts: qcom: pm8150b: add SMB5 charger node

Add the PM8150B SMB5 charger block with its USBIN / ADC IIO channels and
interrupts (usb-plugin, bat-ov, wdog-bark, usbin-icl-change). Left disabled
here; the board DTS enables it.

Signed-off-by: Kumar-Jy <20044626+Kumar-Jy@users.noreply.github.com>
---
 arch/arm64/boot/dts/qcom/pm8150b.dtsi | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

--- a/arch/arm64/boot/dts/qcom/pm8150b.dtsi	2026-08-31 08:32:18.430956781 +0530
+++ b/arch/arm64/boot/dts/qcom/pm8150b.dtsi	2026-08-31 07:13:57.111553815 +0530
@@ -107,6 +107,26 @@
 			#thermal-sensor-cells = <0>;
 		};
 
+		pm8150b_charger: charger@1000 {
+			compatible = "qcom,pm8150b-charger";
+			reg = <0x1000>;
+
+			interrupts = <0x2 0x13 0x4 IRQ_TYPE_EDGE_BOTH>,
+				     <0x2 0x12 0x2 IRQ_TYPE_EDGE_BOTH>,
+				     <0x2 0x16 0x1 IRQ_TYPE_EDGE_RISING>,
+				     <0x2 0x13 0x7 IRQ_TYPE_EDGE_RISING>;
+			interrupt-names = "usb-plugin",
+					  "bat-ov",
+					  "wdog-bark",
+					  "usbin-icl-change";
+
+			io-channels = <&pm8150b_adc ADC5_USB_IN_I>,
+				      <&pm8150b_adc ADC5_USB_IN_V_16>;
+			io-channel-names = "usbin_i", "usbin_v";
+
+			status = "disabled";
+		};
+
 		pm8150b_adc: adc@3100 {
 			compatible = "qcom,spmi-adc5";
 			reg = <0x3100>;
@@ -133,6 +153,18 @@
 				label = "die_temp";
 			};
 
+			channel@7 {
+				reg = <ADC5_USB_IN_I>;
+				qcom,pre-scaling = <1 1>;
+				label = "usbin_i";
+			};
+
+			channel@8 {
+				reg = <ADC5_USB_IN_V_16>;
+				qcom,pre-scaling = <1 16>;
+				label = "usbin_v";
+			};
+
 			channel@9 {
 				reg = <ADC5_CHG_TEMP>;
 				qcom,pre-scaling = <1 1>;
-- 
2.49.0
